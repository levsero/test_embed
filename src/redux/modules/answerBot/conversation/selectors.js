import { createSelector } from 'reselect';
import _ from 'lodash';

import { getSessions } from 'src/redux/modules/answerBot/sessions/selectors';

const getState = state => state.answerBot;
const getConversation = state => state.answerBot.conversation;
const getGroupMessages = (state, props) => props.messageKeys.map(key => state.answerBot.messages.get(key));
const appendGroup = (groups, group) => {
  groups[Object.keys(groups).length] = group;
};

export const getMessages = createSelector(
  [getState],
  state => state.messages
);

export const getLastMessage = createSelector(
  getMessages,
  (messages) => {
    const size = messages.size;

    return size  === 0 ? null : Array.from(messages)[size-1][1];
  }
);

export const getLastMessageType = createSelector(
  getLastMessage,
  (message) => _.get(message, 'type', '')
);

// Group consecutive messages by isVisitor.
// Only store the message keys
export const getMessageGroupKeys = createSelector(
  getMessages,
  (groupMessages) => {
    let groups = {};
    let currentGroup = { messageKeys: [], isVisitor: undefined };
    let feedbackMode = false;
    let feedbackChain = [];
    let lastEntry;

    for (let entry of groupMessages) {
      const message = entry[1];

      lastEntry = entry;
      if (message.type === 'botTyping') {
        continue;
      }

      if (feedbackMode) {
        if (message.type === 'feedback' || message.feedbackRelated) {
          feedbackChain.push(entry);
          continue;
        } else if (message.type !== 'feedbackRequested') {
          currentGroup = appendFeedback(feedbackChain, currentGroup, groups);
          feedbackMode = false;
          feedbackChain = [];
        }
      }

      if (message.type === 'feedbackRequested') {
        feedbackMode = true;
        feedbackChain = [];
        continue;
      }

      currentGroup = updateGroups(currentGroup, groups, entry);
    }

    if (feedbackChain.length > 0) {
      currentGroup = appendFeedback(feedbackChain, currentGroup, groups);
    }
    if (lastEntry && lastEntry[1].type === 'botTyping') {
      currentGroup = updateGroups(currentGroup, groups, lastEntry);
    }
    appendGroup(groups, currentGroup);
    return groups;
  }
);

const userHasResponded = (message) => {
  return (message && message[1].feedbackRelated && message[1].isVisitor);
};

const appendFeedback = (feedbackChain, currentGroup, groups) => {
  _.forEach(feedbackChain, (msg, index) => {
    const current = msg[1];

    if (current.type === 'feedback') {
      const next = feedbackChain[index + 1];

      if (userHasResponded(next)) {
        return;
      }
    }
    currentGroup = updateGroups(currentGroup, groups, msg);
  });

  return currentGroup;
};

const updateGroups = (currentGroup, groups, entry) => {
  const message = entry[1];

  // If the message isVisitor state is different from the current group isVisitor state,
  // create a new group for that state
  if (message.isVisitor !== currentGroup.isVisitor) {
    if (currentGroup.messageKeys.length > 0) {
      appendGroup(groups, currentGroup);
    }
    currentGroup = {
      messageKeys: [],
      isVisitor: message.isVisitor
    };
  }
  currentGroup.messageKeys.push(entry[0]);

  return currentGroup;
};

export const makeGetGroupMessages = () => createSelector(
  [getGroupMessages, getSessions],
  (messages, sessions) => {
    const messagesWithArticle = messages.map(m => {
      if (m.type === 'results') { m.articles = sessions.get(m.sessionID).articles; }
      return m;
    });

    return messagesWithArticle;
  }
);

export const getLastScroll = createSelector(
  [getConversation],
  conversation => conversation.lastScroll
);

export const getLastScreenClosed = createSelector(
  [getConversation],
  conversation => conversation.lastScreenClosed
);
