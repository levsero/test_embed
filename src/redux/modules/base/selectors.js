export const getZopimChatEmbed = (state) => state.base.embeds.zopimChat;
export const getHelpCenterEmbed = (state) => state.base.embeds.helpCenterForm;
