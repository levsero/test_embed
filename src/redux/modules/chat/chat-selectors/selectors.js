import _ from 'lodash'

const chatState = state => state.chat

export const getNotification = state => chatState(state).notification
export const getChats = state => chatState(state).chats
export const isAgent = nick => (nick ? nick.indexOf('agent:') > -1 : false)
export const getThemeMessageType = state => chatState(state).accountSettings.theme.message_type
export const getOrderedAgents = state => chatState(state).agents
export const getShowOperatingHours = state =>
  chatState(state).accountSettings.operatingHours.display_notice
export const getForcedStatus = state => chatState(state).forcedStatus
export const getInactiveAgents = state => chatState(state).inactiveAgents
export const getSocialLogin = state => chatState(state).socialLogin
export const getConnection = state => chatState(state).connection
export const getCurrentMessage = state => chatState(state).currentMessage
export const getChatRating = state => chatState(state).rating
export const getChatScreen = state => chatState(state).screen
export const getShowChatHistory = state => chatState(state).showChatHistory
export const getChatStatus = state => chatState(state).account_status
export const getChatVisitor = state => chatState(state).visitor
export const getIsChatting = state => chatState(state).is_chatting
export const getNotificationCount = state => getNotification(state).count
export const getPostchatFormSettings = state => chatState(state).accountSettings.postchatForm
export const getEmailTranscript = state => chatState(state).emailTranscript
export const getAttachmentsEnabled = state => chatState(state).accountSettings.attachments.enabled
export const getRatingSettings = state => chatState(state).accountSettings.rating
export const getQueuePosition = state => chatState(state).queuePosition
export const getUserSoundSettings = state => chatState(state).userSettings.sound
export const getReadOnlyState = state => chatState(state).formState.readOnlyState
export const getChatOfflineForm = state => chatState(state).formState.offlineForm
export const getOfflineMessage = state => chatState(state).offlineMessage
export const getPreChatFormState = state => chatState(state).formState.preChatForm
export const getEditContactDetails = state => chatState(state).editContactDetails
export const getMenuVisible = state => chatState(state).menuVisible
export const getAgentJoined = state => chatState(state).agentJoined
export const getLastReadTimestamp = state => chatState(state).lastReadTimestamp
export const getOperatingHours = state => chatState(state).operatingHours
export const getLoginSettings = state => chatState(state).accountSettings.login
export const getStandaloneMobileNotificationVisible = state =>
  chatState(state).standaloneMobileNotificationVisible
export const getIsAuthenticated = state => chatState(state).isAuthenticated
export const getZChatVendor = state => chatState(state).vendor.zChat
export const getSliderVendor = state => chatState(state).vendor.slider
export const getWindowSettings = state => chatState(state).accountSettings.chatWindow
export const getThemeColor = state => ({
  base: chatState(state).accountSettings.theme.color.primary,
  text: undefined
})
export const getBadgeColor = state => chatState(state).accountSettings.theme.color.banner
export const getChatAccountSettingsConcierge = state => chatState(state).accountSettings.concierge
export const getChatAccountSettingsOfflineForm = state =>
  chatState(state).accountSettings.offlineForm
export const getChatAccountSettingsPrechatForm = state =>
  chatState(state).accountSettings.prechatForm
export const getDepartments = state => chatState(state).departments
export const getAccountSettingsLauncherBadge = state => chatState(state).accountSettings.banner
export const getChatBadgeEnabled = state => getAccountSettingsLauncherBadge(state).enabled
export const getHideBranding = state => chatState(state).accountSettings.branding.hide_branding
export const getAccountDefaultDepartmentId = state => chatState(state).defaultDepartment.id
export const getDepartmentsList = state => _.values(getDepartments(state))
export const getIsLoggingOut = state => chatState(state).isLoggingOut
export const getChatLog = state => state.chat.chatLog.groups
export const getFirstMessageTimestamp = state => {
  const first = getChats(state)
    .values()
    .next().value

  return first ? first.timestamp : Date.now()
}

export const getCanShowOnlineChat = state => {
  const isChatting = getIsChatting(state)
  const isOnline = getChatStatus(state) === 'online'

  return isChatting || isOnline
}

export const getChatOnline = state => {
  const forcedStatus = getForcedStatus(state)

  if (forcedStatus === 'online') {
    return true
  } else if (forcedStatus === 'offline') {
    return false
  }

  return _.includes(['online', 'away'], getChatStatus(state))
}

export const getThemePosition = state => {
  const position = state.chat.accountSettings.theme.position

  switch (position) {
    case 'br':
      return 'right'
    case 'bl':
      return 'left'
    default:
      return undefined
  }
}

export const getFirstVisitorMessage = state => state.chat.chatLog.firstVisitorMessage
export const getLatestRatingRequest = state => state.chat.chatLog.latestRatingRequest
export const getLatestRating = state => state.chat.chatLog.latestRating
export const getLastMessageAuthor = state => state.chat.chatLog.lastMessageAuthor
export const getLatestAgentLeaveEvent = state => state.chat.chatLog.latestAgentLeaveEvent
export const getLatestQuickReplyKey = state => state.chat.chatLog.latestQuickReply
export const getChatBanned = state => state.chat.chatBanned

export const getHasBackfillCompleted = state => state.chat.chatLogBackfillCompleted
