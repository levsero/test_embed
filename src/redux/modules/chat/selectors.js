import _ from 'lodash';
import { createSelector } from 'reselect';

const getNotification = (state) => state.chat.notification;
const getAgents = (state) => state.chat.agents;
const getChats = (state) => {
  return _.filter([...state.chat.chats.values()], (e) => e.type === 'chat.msg');
};
const getChatsByAgent = (state) => {
  const chats = _.filter(getChats(state), (chat) => _.includes(chat.nick, 'agent'));

  return _.groupBy(chats, (chat) => chat.nick);
};
const getPrechatSettings = (state) => state.chat.accountSettings.prechatForm;

export const getChatNotification = createSelector(
  [getNotification, getAgents, getChatsByAgent],
  (notification, agents, chats) => {
    const { nick } = notification;
    const agentChats = chats[nick];
    const proactive = !!(agentChats && agentChats.length === 1);

    return {
      ...notification,
      ...agents[notification.nick],
      proactive
    };
  }
);

export const getPrechatFormFields = createSelector(
  [getPrechatSettings],
  (prechatSettings) => {
    const { form } = prechatSettings;

    return _.keyBy(_.values(form), 'name');
  }
);
