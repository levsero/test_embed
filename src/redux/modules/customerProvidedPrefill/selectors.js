export const getLastUpdateAcknowledged = (state, type, id) =>
  state.customerProvidedPrefill.acknowledged[type]?.[id]
export const getLastTimestamp = (state, type) =>
  state.customerProvidedPrefill.types[type]?.timestamp
export const getValues = (state, type) => state.customerProvidedPrefill.types[type]?.values
