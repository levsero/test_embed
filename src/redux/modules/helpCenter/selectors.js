export const getSearchLoading = (state) => state.helpCenter.searchLoading;
export const getArticleClicked = (state) => state.helpCenter.articleClicked;
export const getSearchFailed = (state) => state.helpCenter.searchFailed;
export const getSearchTerm = (state) => state.helpCenter.searchTerm.current;
export const getPreviousSearchTerm = (state) => state.helpCenter.searchTerm.previous;
export const getHasSearched = (state) => state.helpCenter.hasSearched;
export const getHasContextuallySearched = (state) => state.helpCenter.hasContextuallySearched;
