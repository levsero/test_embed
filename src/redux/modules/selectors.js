import { createSelector } from 'reselect';

import { getShowOfflineChat,
  getOfflineFormEnabled,
  getStandaloneMobileNotificationVisible } from './chat/chat-selectors';
import { getZopimChatOnline } from './zopimChat/zopimChat-selectors';
import { getSettingsChatSuppress } from './settings/settings-selectors';
import { getEmbeddableConfigEnabled, getAgentAvailability } from './talk/talk-selectors';
import { getActiveTicketForm, getTicketForms } from './submitTicket/submitTicket-selectors';
import { getActiveEmbed,
  getHelpCenterEmbed,
  getSubmitTicketEmbed,
  getZopimChatEmbed,
  getTalkEmbed,
  getChatEmbed as getNewChatEmbed,
  getIPMWidget } from './base/base-selectors';

import { settings } from 'service/settings';
import { getHasSearched,
  getContextualHelpRequestNeeded,
  getArticleViewActive } from './helpCenter/helpCenter-selectors';
import { isMobileBrowser } from 'utility/devices';

import { MAX_WIDGET_HEIGHT_NO_SEARCH, WIDGET_MARGIN } from 'src/constants/shared';
/*
 * Terms:
 * Available: When an embed is part of config, not suppressed and has all the conditions to be used.
 * Enabled: When an embed is part of config, not suppressed but does not have all the conditions to be used
 */

const getChatEmbed = (state) => getNewChatEmbed(state) || getZopimChatEmbed(state);
const getWidgetFixedFrameStyles = createSelector(
  [getStandaloneMobileNotificationVisible,
    getHasSearched,
    getContextualHelpRequestNeeded,
    getIPMWidget,
    getArticleViewActive,
    getHelpCenterEmbed],
  (standaloneMobileNotificationVisible,
    hasSearched, contextualHelpRequestNeeded,
    isUsingIPMWidgetOnly,
    articleViewActive,
    helpCenterEnabled) => {
    if (isUsingIPMWidgetOnly) {
      return {};
    }

    const hcSuppressed = settings.get('helpCenter.suppress');

    if (!hcSuppressed && !isMobileBrowser() && helpCenterEnabled && !hasSearched &&
        !contextualHelpRequestNeeded && !articleViewActive) {
      return {
        maxHeight: `${MAX_WIDGET_HEIGHT_NO_SEARCH + WIDGET_MARGIN}px`
      };
    }

    if (standaloneMobileNotificationVisible) {
      return {
        height: '33%',
        background: 'transparent'
      };
    }
    return {};
  }
);

export const getChatOnline = (state) => getZopimChatOnline(state) || !getShowOfflineChat(state);

export const getChatEnabled = (state) => getChatEmbed(state) && !getSettingsChatSuppress(state);

export const getChatOfflineAvailable = (state) => getChatEnabled(state) &&
  !getChatOnline(state) && getNewChatEmbed(state) && getOfflineFormEnabled(state) && !getSubmitTicketEmbed(state);

export const getChatAvailable = (state) => getChatEnabled(state) && getChatOnline(state);
export const getShowTalkBackButton = (state) => {
  return getHelpCenterEmbed(state) || getChatAvailable(state) || getSubmitTicketEmbed(state);
};
export const getTalkEnabled = (state) => getTalkEmbed(state) && getEmbeddableConfigEnabled(state);
export const getTalkAvailable = (state) => getTalkEnabled(state) && getAgentAvailability(state);
export const getShowTicketFormsBackButton = createSelector(
  [getActiveTicketForm, getTicketForms, getActiveEmbed],
  (activeForm, ticketForms, activeEmbed) => {
    return activeForm && ticketForms.length > 1 && activeEmbed === 'ticketSubmissionForm';
  }
);

export const getFixedStyles = (state, frame = 'webWidget') => {
  if (frame === 'webWidget') {
    return getWidgetFixedFrameStyles(state);
  }
  return {};
};

export const getIsOnInitialDesktopSearchScreen = (state) => {
  return !!getFixedStyles(state, 'webWidget').maxHeight;
};

export const getMaxWidgetHeight = (state, frame = 'webWidget') => {
  const fixedStyles = getFixedStyles(state, frame);

  if (getIsOnInitialDesktopSearchScreen(state) && fixedStyles.maxHeight) {
    return parseInt(fixedStyles.maxHeight) - WIDGET_MARGIN;
  }

  return undefined;
};
