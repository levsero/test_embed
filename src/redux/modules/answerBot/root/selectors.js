import _ from 'lodash'

import { createSelector } from 'reselect'
import { getSessions } from '../sessions/selectors'
import {
  getHasContextuallySearched,
  getSearchLoading,
  getResultsCount,
  getArticles
} from 'src/redux/modules/helpCenter/helpCenter-selectors'
import {
  getLastMessageType,
  getGetInTouchVisible
} from 'src/redux/modules/answerBot/conversation/selectors'
import { getChannelAvailable } from 'src/redux/modules/selectors'

const getState = state => state.answerBot

export const getCurrentSessionID = createSelector(
  [getState],
  state => state.currentSessionID
)

export const getQuestionValueChangedTime = createSelector(
  [getState],
  state => state.questionValueChangedTime
)

export const getCurrentRequestStatus = createSelector(
  [getCurrentSessionID, getSessions],
  (sessionID, sessions) => {
    const session = sessions && sessions.get(sessionID)

    return session ? session.requestStatus : null
  }
)

export const getCurrentDeflection = createSelector(
  [getCurrentSessionID, getSessions],
  (sessionID, sessions) => {
    const session = sessions && sessions.get(sessionID)

    return session ? session.deflection : null
  }
)

export const getCurrentInteractionToken = createSelector(
  [getCurrentSessionID, getSessions],
  (sessionID, sessions) => {
    const session = sessions && sessions.get(sessionID)

    return session ? session.interactionToken : null
  }
)

export const getCurrentQuery = createSelector(
  [getCurrentSessionID, getSessions],
  (sessionID, sessions) => {
    const session = sessions && sessions.get(sessionID)

    return session ? session.query : null
  }
)

export const isCurrentSessionResolved = createSelector(
  [getCurrentSessionID, getSessions],
  (sessionID, sessions) => {
    const session = sessions && sessions.get(sessionID)

    return !!(session && session.resolved)
  }
)

export const getCurrentArticleID = createSelector(
  getState,
  state => _.get(state, 'currentArticle.articleID', null)
)

const getCurrentContextualArticleID = createSelector(
  getState,
  state => _.get(state, 'currentContextualArticle.articleID', null)
)

export const getCurrentArticleSessionID = createSelector(
  [getState],
  state => (state.currentArticle ? state.currentArticle.sessionID : null)
)

const getCurrentContextualArticle = createSelector(
  [getCurrentContextualArticleID, getArticles],
  (articleID, articles) => (articleID ? _.find(articles, { id: articleID }) : null)
)

export const getCurrentArticle = createSelector(
  [getCurrentContextualArticle, getCurrentArticleID, getCurrentArticleSessionID, getSessions],
  (contextualArticle, articleID, sessionID, sessions) => {
    if (contextualArticle) return contextualArticle
    const session = sessions && sessions.get(sessionID)
    const articles = (session && session.articles) || []

    return _.find(articles, { id: articleID })
  }
)

export const isFeedbackRequired = createSelector(
  [
    getCurrentContextualArticleID,
    getCurrentArticle,
    getCurrentArticleSessionID,
    getCurrentSessionID,
    isCurrentSessionResolved
  ],
  (contextual, article, sessionID, currentSessionID, currentSessionResolved) => {
    return (
      !contextual &&
      sessionID === currentSessionID &&
      !currentSessionResolved &&
      !(article && article.markedAsIrrelevant)
    )
  }
)

export const getCurrentScreen = createSelector(
  [getState],
  state => state.currentScreen
)

export const getCurrentMessage = createSelector(
  [getState],
  state => state.currentMessage
)

export const getGreeted = createSelector(
  [getState],
  state => !!state.greeted
)

export const getInitialFallbackSuggested = createSelector(
  [getState],
  state => !!state.initialFallbackSuggested
)

export const getContextualSearchFinished = createSelector(
  getState,
  state => !!state.contextualSearchFinished
)

export const getContextualSearchStatus = createSelector(
  [getHasContextuallySearched, getSearchLoading, getLastMessageType, getResultsCount],
  (hasContextuallySearched, searchLoading, lastMessageType, resultsCount) => {
    if (!hasContextuallySearched) return null
    if (searchLoading && lastMessageType !== 'botTyping') return 'PENDING'
    if (!searchLoading && resultsCount > 0) return 'COMPLETED'
    if (!searchLoading && resultsCount === 0) return 'NO_RESULTS'
    return null
  }
)

export const getContactButtonVisible = createSelector(
  [getGetInTouchVisible, getChannelAvailable],
  (getInTouchVisible, channelAvailable) => channelAvailable && getInTouchVisible
)
