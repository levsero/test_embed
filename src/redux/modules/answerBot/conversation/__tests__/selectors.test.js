import * as selectors from '../selectors';

describe('getMessages', () => {
  it('returns the messages', () => {
    const mockState = {
      answerBot: {
        messages: new Map([
          ['1', { message: 'Thing 1' }],
          ['2', { message: 'Thing 2' }]
        ])
      }
    };
    const result = selectors.getMessages(mockState);

    expect(result)
      .toMatchSnapshot();
  });
});

describe('getGroupMessages', () => {
  it('returns messages according to passed in keys', () => {
    const mockState = {
      answerBot: {
        messages: new Map([[0, 'zero'], [1, 'one'], [2, 'two']])
      }
    };
    const mockProps = {
      messageKeys: [1, 2]
    };
    const results = selectors.makeGetGroupMessages(mockState, mockProps);

    expect(results(mockState, mockProps))
      .toMatchSnapshot();
  });
});

describe('getLastScroll', () => {
  it('returns conversation lastScroll', () => {
    const results = selectors.getLastScroll.resultFunc({ lastScroll: 123 });

    expect(results)
      .toMatchSnapshot();
  });
});

describe('getLastScreenClosed', () => {
  it('returns conversation lastScreenClosed', () => {
    const results = selectors.getLastScreenClosed.resultFunc({ lastScreenClosed: 123 });

    expect(results)
      .toMatchSnapshot();
  });
});

describe('makeGetGroupMessages', () => {
  it('returns messages with articles embedded', () => {
    const mockState = {
      answerBot: {
        messages: new Map(
          [
            ['message1', { type: 'results', sessionID: '123' }]
          ]
        ),
        sessions: new Map([['123', { resolved: true, articles: [{ id:1 }, { id: 2 }] }]])
      }
    };
    const mockProps = {
      messageKeys: ['message1']
    };
    const results = selectors.makeGetGroupMessages(mockState, mockProps);

    expect(results(mockState, mockProps))
      .toMatchSnapshot();
  });
});

describe('getGroupMessageKeys', () => {
  it('groups message keys by isVisitor', () => {
    const mockState = {
      answerBot: {
        messages: new Map(
          [
            [0, { }],
            [1, { }],
            [2, { isVisitor: true }],
            [3, { }]
          ]
        )
      }
    };
    const results = selectors.getMessageGroupKeys(mockState);

    expect(results)
      .toMatchSnapshot();
  });

  it('discards previous feedback', () => {
    const mockState = {
      answerBot: {
        messages: new Map(
          [
            [0, { type: 'feedbackRequested' }],
            [1, { feedbackRelated: true }],
            [2, { type: 'feedbackRequested' }],
            [3, { feedbackRelated: true }]
          ]
        )
      }
    };
    const results = selectors.getMessageGroupKeys(mockState);

    expect(results)
      .toMatchSnapshot();
  });

  it('handles feedback appropriately', () => {
    const mockState = {
      answerBot: {
        messages: new Map(
          [
            [0, {}],
            [1, { isVisitor: true } ],
            [2, { type: 'results' } ],
            [3, { type: 'feedbackRequested' }], // start feedback session
            [4, { feedbackRelated: true }],
            [5, { type: 'feedback', isVisitor: true }],
            [6, { feedbackRelated: true }],
            [7, { type: 'feedbackRequested' }], // new feedback session, discard previous ones
            [8, { feedbackRelated: true }],
            [9, { isVisitor: true } ] // non-feedback related message, record current feedback
          ]
        )
      }
    };
    const results = selectors.getMessageGroupKeys(mockState);

    expect(results)
      .toMatchSnapshot();
  });
});
