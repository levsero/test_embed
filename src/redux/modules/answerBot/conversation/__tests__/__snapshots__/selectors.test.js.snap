// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getGroupMessageKeys discards previous feedback 1`] = `
Object {
  "0": Object {
    "isVisitor": undefined,
    "messageKeys": Array [
      3,
    ],
  },
}
`;

exports[`getGroupMessageKeys groups message keys by isVisitor 1`] = `
Object {
  "0": Object {
    "isVisitor": undefined,
    "messageKeys": Array [
      0,
      1,
    ],
  },
  "1": Object {
    "isVisitor": true,
    "messageKeys": Array [
      2,
    ],
  },
  "2": Object {
    "isVisitor": undefined,
    "messageKeys": Array [
      3,
    ],
  },
}
`;

exports[`getGroupMessageKeys handles feedback appropriately 1`] = `
Object {
  "0": Object {
    "isVisitor": undefined,
    "messageKeys": Array [
      0,
    ],
  },
  "1": Object {
    "isVisitor": true,
    "messageKeys": Array [
      1,
    ],
  },
  "2": Object {
    "isVisitor": undefined,
    "messageKeys": Array [
      2,
      8,
    ],
  },
  "3": Object {
    "isVisitor": true,
    "messageKeys": Array [
      9,
    ],
  },
}
`;

exports[`getGroupMessages returns messages according to passed in keys 1`] = `
Array [
  "one",
  "two",
]
`;

exports[`getLastScreenClosed returns conversation lastScreenClosed 1`] = `123`;

exports[`getLastScroll returns conversation lastScroll 1`] = `123`;

exports[`getMessages returns the messages 1`] = `
Map {
  "1" => Object {
    "message": "Thing 1",
  },
  "2" => Object {
    "message": "Thing 2",
  },
}
`;

exports[`makeGetGroupMessages returns messages with articles embedded 1`] = `
Array [
  Object {
    "articles": Array [
      Object {
        "id": 1,
      },
      Object {
        "id": 2,
      },
    ],
    "sessionID": "123",
    "type": "results",
  },
]
`;
